### PR TITLE
Project build tools update

### PR DESCRIPTION
**Pull request by mjtrangoni**
Hi @ofesseler , @coder-hugo ,

I did some refactoring and quality-assurance work today. PTAL.

* Add [gometalinter](https://github.com/alecthomas/gometalinter) lint tools.
* Switch from `govendor` to [dep](https://github.com/golang/dep). This is a much more powerful dependency management tool.
* Switch from `flag` to [kingpin](https://github.com/alecthomas/kingpin) as many Prometheus tools.
* Add a `CHANGELOG.md` file.
* Update `README.md` file.
* Update **TravisCI** configuration, adding gometalinter checks.

Thanks!

### CODE REVIEW

**Comment by mjtrangoni** [comment]
well, TravisCI is ok now.

File: Makefile [hunk 1]
```diff
@@ -1,28 +1,40 @@
-GO = go
-pkgs = $(shell $(GO) list ./... | grep -v /vendor/)
-PROMU = promu
-BIN_DIR ?= $(shell pwd)
+GO           ?= GO15VENDOREXPERIMENT=1 go
```
Comment (ofesseler) [inline Makefile]: Is it for backwards compatibility?

File: main.go [hunk 2]
```diff
@@ -306,8 +306,9 @@ func (e *Exporter) Collect(ch chan<- prometheus.Metric) {
 	}
 	for _, vol := range volumeStatusAll.VolStatus.Volumes.Volume {
 		for _, node := range vol.Node {
-			if node.Status != 1 {
-			}
+			//FIXME: what is this intended for?
```
Comment (ofesseler) [inline main.go]: Don't remember why I made this. Just delete it. right now it is useless!

File: Makefile (hunk 1 above)
Comment (mjtrangoni) [reply Makefile]: yep, we don`t need it anymore. See,
> Go 1.5 introduced experimental support for a "vendor" directory, enabled by the > GO15VENDOREXPERIMENT environment variable. Go 1.6 enabled this behavior by default, and in Go 1.7, this switch has been removed and the "vendor" behavior is > always enabled.
I will remove this tomorrow in another PR.

**Comment by mjtrangoni** [comment]
Thanks!

**Pull request merged**